internal/yggdrasil/configuration-assembler.go

```diff
@@ -0,0 +1,475 @@
+package yggdrasil
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/project-flotta/flotta-operator/internal/k8sclient"
+	"github.com/project-flotta/flotta-operator/internal/storage"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/project-flotta/flotta-operator/internal/configmaps"
+	"github.com/project-flotta/flotta-operator/internal/images"
+	"github.com/project-flotta/flotta-operator/internal/repository/edgedeployment"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/ghodss/yaml"
+	"github.com/go-logr/logr"
+	"github.com/project-flotta/flotta-operator/api/v1alpha1"
+	"github.com/project-flotta/flotta-operator/internal/devicemetrics"
+	"github.com/project-flotta/flotta-operator/internal/labels"
+	"github.com/project-flotta/flotta-operator/models"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type configurationAssembler struct {
+	allowLists             devicemetrics.AllowListGenerator
+	claimer                *storage.Claimer
+	client                 k8sclient.K8sClient
+	configMaps             configmaps.ConfigMap
+	deploymentRepository   edgedeployment.Repository
+	recorder               record.EventRecorder
+	registryAuthRepository images.RegistryAuthAPI
+}
+
+func (a *configurationAssembler) getDeviceConfiguration(ctx context.Context, edgeDevice *v1alpha1.EdgeDevice, logger logr.Logger) (*models.DeviceConfigurationMessage, error) {
+	var workloadList models.WorkloadList
+	var secretList models.SecretList
+	if edgeDevice.DeletionTimestamp == nil {
+		var edgeDeployments []v1alpha1.EdgeDeployment
+
+		for _, deployment := range edgeDevice.Status.Deployments {
+			edgeDeployment, err := a.deploymentRepository.Read(ctx, deployment.Name, edgeDevice.Namespace)
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					logger.Error(err, "cannot retrieve Edge Deployments")
+					return nil, err
+				}
+				continue
+			}
+			if edgeDeployment.DeletionTimestamp == nil {
+				edgeDeployments = append(edgeDeployments, *edgeDeployment)
+			}
+		}
+
+		wl, err := a.toWorkloadList(ctx, logger, edgeDeployments, edgeDevice)
```

Comment:
can `wl` be replaced with `workloadList`?

Reply:
Done.